library/src/scala/quoted/Quotes.scala

```diff
@@ -3807,6 +3883,19 @@ trait Quotes { self: runtime.QuoteUnpickler & runtime.QuoteMatching =>
 
         /** Case class or case object children of a sealed trait or cases of an `enum`. */
         def children: List[Symbol]
+
+        /** Type reference to the symbol usable in the scope of its owner.
+         *
+         *  To get a reference to a symbol from a specific prefix `tp`, use `tp.memberType(symbol)` instead.
+         *
+         *  @pre symbol.isType returns true
+         */
+        @experimental
+        def typeRef: TypeRef
```

Comment:
It seems that we're missing an API to create a TypeRef/TermRef from a specific prefix: with `foo.typeRef` I can get a TypeRef `this.foo` and with `pre.memberType(foo)` I can get the info corresponding to `pre.foo`, but is there a way to generate the TypeRef `pre.foo` itself?

Reply:
Doesn't `TypeRepr.select` provide that?

Reply:
Oh indeed, I missed that one, maybe each of their documentation should `@see` the other one.

Reply:
Done